### PR TITLE
Accessibility changes

### PR DESCRIPTION
**Pull request by dancgray**
Need to add aria labels to course.json:
                "pageLevelProgress": "Page navigation with component completion status",
                "pageLevelProgressIndicatorBar": "Page level progress bar. You have completed"

### CODE REVIEW

**Comment by oliverfoster** [comment]
#### menu template

Current:
```
<div class="page-level-progress-menu-item-indicator">
    <div class="page-level-progress-menu-item-indicator-bar" style="width:{{completedChildrenAsPercentage}}%" role="region" aria-label="" {{#if _globals._accessibility._ariaLabels.pageLevelProgressIndicatorBar}}tabindex="0"{{/if}}></div>
</div>
```

Suggested:
```
<div class="page-level-progress-menu-item-indicator">
    <div class="page-level-progress-menu-item-indicator-bar" style="width:{{completedChildrenAsPercentage}}%" role="region" tabindex="0"></div>
</div>
```

You can remove the if statement and the blank aria-label tag. The aria-label is added dynamically and the element always needs to be tabbable.

**Comment by oliverfoster** [comment]
#### drawer template - pageLevelProgress.hbs

Current:
```
<div class="page-level-progress-inner" role="region" aria-label="{{_globals._accessibility._ariaLabels.pageLevelProgress}}" {{#if _globals._accessibility._ariaLabels.pageLevelProgress}}tabindex="0"{{/if}}>
    {{#each components}}
    <div class="page-level-progress-item drawer-item">
        {{#if _isVisible}}
        <a class="page-level-progress-item-title clearfix drawer-item-open" tabindex="0" role="button" href="#" data-page-level-progress-id="{{_id}}">
        {{else}}
        <div class="page-level-progress-item-title drawer-item-open disabled clearfix">
        {{/if}}
            <div class="page-level-progress-item-title-inner accessible-text-block h5" role="heading" aria-level="5" tabindex="0">
            {{{title}}}
            </div>
            <div class="page-level-progress-indicator page-level-progress-indicator-{{#if _isComplete}}complete{{else}}incomplete{{/if}}" role="region" aria-label="{{#if _isComplete}}{{_globals._accessibility._ariaLabels.complete}}{{else}}{{_globals._accessibility._ariaLabels.incomplete}}{{/if}}" tabindex="0">
                <div class="page-level-progress-indicator-bar">
                </div>
            </div>
        {{#if _isVisible}}
        </a>
        {{else}}
        </div>
        {{/if}}
    </div>
    {{/each}}
</div>www
```

Suggested:
```
<div class="page-level-progress-inner" role="region" aria-label="{{_globals._accessibility._ariaLabels.pageLevelProgress}}" {{#if _globals._accessibility._ariaLabels.pageLevelProgress}}tabindex="0"{{/if}}>
    <div role="list">
    {{#each components}}
    <div role="listitem" class="page-level-progress-item drawer-item">
        {{#if _isVisible}}
        <a class="page-level-progress-item-title clearfix drawer-item-open" tabindex="0" role="button" href="#" data-page-level-progress-id="{{_id}}" aria-label="{{title}} {{#if _isComplete}}{{_globals._accessibility._ariaLabels.complete}}{{else}}{{_globals._accessibility._ariaLabels.incomplete}}{{/if}}">
        {{else}}
        <div class="page-level-progress-item-title drawer-item-open disabled clearfix">
        {{/if}}
            <div class="page-level-progress-item-title-inner accessible-text-block h5">
            {{{title}}}
            </div>
            <div class="page-level-progress-indicator page-level-progress-indicator-{{#if _isComplete}}complete{{else}}incomplete{{/if}}">
                <div class="page-level-progress-indicator-bar">
                </div>
            </div>
        {{#if _isVisible}}
        </a>
        {{else}}
        </div>
        {{/if}}
    </div>
    {{/each}}
    </div>
</div>
```

correction:
Remove the 'www' from the end line 23.

lists:
Add a role='list' wrapper to the items list
Add a role='listitem' tag to each item

data presentation:
Remove tabbable index from .page-level-progress-item-title-inner
Remove the aria label from .page-level-progress-indicator
Add an aria-label to listitem tag instead, using the listitem's title + complete/incomplete status.
This will make each listitem have only one tabbable region containing all of the relevant information.

**Comment by oliverfoster** [comment]
#### drawer view - PageLevelProgressView.js

Current:
```
        render: function() {
            var data = this.collection.toJSON();
            var template = Handlebars.templates['pageLevelProgress'];
            this.$el.html(template({components: data}));
            return this;
        }
```

Suggested:
```
        render: function() {
            var components = this.collection.toJSON();
            var data = _.extend({}, {
                components: components,
                _globals: Adapt.course.get('_globals')
            });
            var template = Handlebars.templates['pageLevelProgress'];
            this.$el.html(template(data));
            this.$el.a11y_aria_label(true);
            return this;
        }
```

Template variables:
Reconstruct the template variables to contain the _globals object.

Aria Labels:
Make sure all aria_labels are handled by query.a11y, otherwise you'll have nested regions - which some screen readers don't like.

**Comment by oliverfoster** [comment]
#### menu view - PageLevelProgressNavigationView.js

Current:
```
        render: function() {
            var data = this.collection.toJSON();
            var template = Handlebars.templates['pageLevelProgressNavigation'];
            $('.navigation-drawer-toggle-button').after(this.$el.html(template({components: data})));
            return this;
        },
```

Suggested:
```
        render: function() {
            var components = this.collection.toJSON();
            var data = _.extend({}, {
                components: components,
                _globals: Adapt.course.get('_globals')
            });
            var template = Handlebars.templates['pageLevelProgressNavigation'];
            $('.navigation-drawer-toggle-button').after(this.$el.html(template(data)));
            return this;
        },
```

Template variables:
Reconstruct the template variables to contain the _globals object.

**Comment by oliverfoster** [comment]
#### menu view - PageLevelProgressMenuView.js

Current:
```
        render: function() {
            var data = this.model.toJSON();
            var template = Handlebars.templates['pageLevelProgressMenu'];
            this.$el.html(template(data));
            return this;
        },
```

Suggested:
```
        render: function() {
            var data = this.model.toJSON();
            _.extend(data, {
                _globals: Adapt.course.get('_globals')
            });
            var template = Handlebars.templates['pageLevelProgressMenu'];
            this.$el.html(template(data));
            return this;
        },
```

Template variables:
Reconstruct the template variables to contain the _globals object.

**Comment by oliverfoster** [comment]
### course.json

Submit a PR containing something like:

```
                "pageLevelProgress": "Page sections",
                "pageLevelProgressIndicatorBar": "You have completed "
```
In _globals._accessibility._ariaLabels

**Comment by dancgray** [comment]
Applied suggestions, Also added a11y_focus() to scrollToPageElement.

File: js/PageLevelProgressView.js [hunk 1]
```diff
@@ -21,14 +21,20 @@ define(function(require) {
             var currentComponentSelector = '.' + $(event.currentTarget).attr('data-page-level-progress-id');
             var $currentComponent = $(currentComponentSelector);
             $(window).scrollTo($currentComponent, {offset: {top: -$('.navigation').height()}});
+            $currentComponent.a11y_focus();
```
Comment (oliverfoster) [inline js/PageLevelProgressView.js]: should be:

**Comment by dancgray** [comment]
Good spot, thanks Ollie.

**Comment by oliverfoster** [comment]
+1 lovely

**Comment by moloko** [comment]
+1

**Comment by taylortom** [comment]
+1

**Pull request merged**